activerecord/lib/active_record/associations/singular_association.rb

```diff
@@ -3,7 +3,7 @@ module Associations
     class SingularAssociation < Association #:nodoc:
       # Implements the reader method, e.g. foo.bar for Foo.has_one :bar
       def reader(force_reload = false)
-        if force_reload
+        if force_reload && klass
```

Comment:
It is entirely possible I'm being dense here, but if the polymorphic
association hasn't set a `klass`, then what is there to reload? Shouldn't
the reader always return `nil` in that case?

On Tue, Jun 16, 2015 at 11:58 AM, Arthur Nogueira Neves <
notifications@github.com> wrote:

> In activerecord/lib/active_record/associations/singular_association.rb
> <https://github.com/rails/rails/pull/20552#discussion_r32536316>:
>
> > @@ -3,7 +3,7 @@ module Associations
> >      class SingularAssociation < Association #:nodoc:
> >        # Implements the reader method, e.g. foo.bar for Foo.has_one :bar
> >        def reader(force_reload = false)
> > -        if force_reload
> > +        if force_reload && klass
>
> I think the problem now, is that instead of getting the error, the
> force_reload will not work for polymorphic. as it will not call reload.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rails/rails/pull/20552/files#r32536316>.
>